editor/src/clj/editor/ui/fuzzy_choices.clj

```diff
@@ -53,10 +54,16 @@
 (defn filter-options [option->matched-text option->label-text filter-text options]
   (if (empty? filter-text)
     options
-    (sort (partial option-order option->label-text)
-          (seq (r/foldcat (r/filter some?
-                                    (r/map (partial option->fuzzy-matched-option option->matched-text filter-text)
-                                           options)))))))
+    (let [fuzzy-matched-options
+          (->> options
+               (r/take-while (thread-util/thread-uninterrupted-predicate))
+               (r/map #(option->fuzzy-matched-option option->matched-text filter-text %))
+               (r/filter some?)
+               (r/foldcat))]
+
+      (thread-util/throw-if-interrupted!)
+      (sort (partial option-order option->label-text)
+            (seq fuzzy-matched-options)))))
```

Comment:
Tidy up formatting and inject interruption checks. Since reducers run on other threads in the thread pool, we use `(thread-util/thread-uninterrupted-predicate)` to check the interrupted flag of the the instigating thread, not the worker thread.